### ISSUE TITLE
how do you turn off TE learning in Lora Module?

### ISSUE THREAD

**Issue by rafstahelin** [bmaltais/kohya_ss#2561]
jUst wondering how to turn off TE learning in Lora Module? In Finetune there's --train_unet_only. But in Lora, how do you accomplish this?

**Comment by b-fission** [comment]
There are 3 learning rates to configure. 

The first one is simply called "Learning rate" but it can be overridden by the other two: "Text Encoder learning rate (optional)" and "Unet learning rate (optional)"

If you want to disable TE training, then ignore the first Learning rate, set the learning rates for Text Encoder to 0 and Unet to whatever value you need.

You can verify that it's training unet-only by looking for the line `network_train_unet_only = true` in the config_lora-xxxxx.toml file.

Another way to verify is through the log output. When you are training unet only, the log should only say `enable` for the Unet but not for the text encoder

```
INFO     create LoRA network. base dim (rank): 64, alpha: 32
INFO     neuron dropout: p=0.1, rank dropout: p=0.1, module dropout: p=0.1
INFO     create LoRA for Text Encoder 1:
INFO     create LoRA for Text Encoder 2:
INFO     create LoRA for Text Encoder: 264 modules.
INFO     create LoRA for U-Net: 722 modules.
INFO     enable LoRA for U-Net: 722 modules
prepare optimizer, data loader etc.
```

**Comment by rafstahelin** [comment]
Oh gosh, the devil in the details. I didn't understand the gui's top LR entry. When does one use the same values for Unet and te. But makes perfect sense to only use the bottom values 
Thanks for clear explanation. I can see now that the te is correctly behaving on the graphs thank you very much for your kind response

**Comment by rafstahelin** [comment]
> There are 3 learning rates to configure. 
> 
> 
> 
> The first one is simply called "Learning rate" but it can be overridden by the other two: "Text Encoder learning rate (optional)" and "Unet learning rate (optional)"
> 
> 
> 
> If you want to disable TE training, then ignore the first Learning rate, set the learning rates for Text Encoder to 0 and Unet to whatever value you need.
> 
> 
> 
> You can verify that it's training unet-only by looking for the line `network_train_unet_only = true` in the config_lora-xxxxx.toml file.
> 
> 
> 
> Another way to verify is through the log output. When you are training unet only, the log should only say `enable` for the Unet but not for the text encoder
> 
> 
> 
> ```
> 
> INFO     create LoRA network. base dim (rank): 64, alpha: 32
> 
> INFO     neuron dropout: p=0.1, rank dropout: p=0.1, module dropout: p=0.1
> 
> INFO     create LoRA for Text Encoder 1:
> 
> INFO     create LoRA for Text Encoder 2:
> 
> INFO     create LoRA for Text Encoder: 264 modules.
> 
> INFO     create LoRA for U-Net: 722 modules.
> 
> INFO     enable LoRA for U-Net: 722 modules
> 
> prepare optimizer, data loader etc.
> 
> ```

@b-fission btw, do you do any consulting?

**Comment by b-fission** [comment]
No, I'm just dabbling with the tools here occasionally.

**Comment by rafstahelin** [comment]
@b-fission thanks no problem
